lib/modules/dosomething/dosomething_reportback/includes/Reportback.php

```diff
@@ -333,27 +333,72 @@ public function deleteFiles() {
    *   Array storing values for flagged/promoted reasons.
    */
   public function setFlaggedPromoted($status = NULL, $values = NULL) {
+    $items = ReportbackItem::get($this->getFids());
+
+    // Determine all of the reportback statuses
+    $flaggedReportbacks = false;
+    $promotedReportbacks = false;
+
+    foreach ($items as $item) {
+      if ($item->status === "flagged") {
+        $flaggedReportbacks = TRUE;
+
+      }
+      else if($item->status === "promoted") {
+        $promotedReportbacks = TRUE;
+      }
+    }
+
+    // Verifies that reportbacks get the correct status and boolean
+    // regardless of the order they are reviewed or how many there is
+    if ($flaggedReportbacks || $status === "flagged") {
+      $status = "flagged";
+    }
+    // Promoted must be after flagged as it has second priority
+    else if($promotedReportbacks || $status === "promoted") {
+      $status = "promoted";
+    }
+
+    // Based on the logic decided above, modify the review values
     if ($status === 'flagged') {
       $this->flagged = 1;
-      $this->flagged_reason = $values['flagged_reason'] ?: NULL;
       $this->promoted = 0;
+
+      // Make sure the reason is correct
+      if (is_string($values['flagged_reason'])) {
+        if (empty($values['flagged_reason'])) {
```

Comment:
is_string verifies that the value isnt an array. If you don't select any option for an item the value in values['flagged_reason'] will be an array containing each option and 0. the second part verifies that the string being sent isnt empty (happens when you just save the page without clicking anything new)